src/slack-notifications/slack-notifications-command.conf

```diff
@@ -65,38 +69,60 @@ object NotificationCommand "slack-notifications-command" {
     }
 
     try {
-        var icon = slack_icon_dictionary.get(notification_type)
+        var icon = slack_icon_dictionary.get(service_state)
```

Comment:
The icon will be then chosen based on the `service_state`, ignoring the `host_state` and the `notification_type`
e.g. In case of the "custom" notification, the icon will still be chosen based on the service_state...is this intended?
maybe we could chose the icon based on the notification _type_ and the color based on the _state_ (respectively service shall take service state and host-> host's)

Reply:
You are right, I missed the host_state being ignored case. But I was testing this and I didn't really like the same icon for both WARNING and CRITICAL alerts, but I guess we can leave it as it was.

>maybe we could chose the icon based on the notification type and the color based on the state (respectively service shall take service state and host-> host's)
I agree.